src/Serializer.php

```diff
@@ -26,13 +26,31 @@ class Serializer implements SerializerInterface, NormalizerInterface, BinarySeri
     private $loader;
 
     /**
-     * Serializer constructor.
-     *
+     * @var DenormalizationContext|null
+     */
+    private $defaultDenormalizationContext;
+
+    /**
+     * @var NormalizationContext|null
+     */
+    private $defaultNormalizationContext;
+
+    /**
      * @param NormalizerLoaderInterface $loader
+     * @param array<string, mixed>|null $defaultDenormalizationOptions Default options to use when denormalizing (i.e. convert serialized data to PHP data).
+     * @param array<string, mixed>|null $defaultNormalizationOptions Default options to use when normalizing (i.e. convert PHP data to serialized data).
      */
-    public function __construct(NormalizerLoaderInterface $loader)
+    public function __construct(NormalizerLoaderInterface $loader, ?array $defaultDenormalizationOptions = null, ?array $defaultNormalizationOptions = null)
     {
         $this->loader = $loader;
+
+        if ($defaultDenormalizationOptions !== null) {
+            $this->defaultDenormalizationContext = new DenormalizationContext($this, $defaultDenormalizationOptions);
+        }
+
+        if ($defaultNormalizationOptions !== null) {
+            $this->defaultNormalizationContext = new NormalizationContext($this, $defaultNormalizationOptions);
+        }
```

Comment:
This will cause a circular reference. We should store the context array and see what we can do on context instantiation